src/lib/ModelConfig.cc

```diff
@@ -421,20 +410,44 @@ void ModelConfig::validate() {
   }
   subFields.clear();
 
-  // LSQ-L1-Interface
-  root = "LSQ-L1-Interface";
-  subFields = {"Access-Latency",
-               "Exclusive",
+  // Memory-Hierarchy
+  root = "Memory-Hierarchy";
+  subFields = {"Cache-Line-Width", "L1-Data", "DRAM"};
+  std::vector<std::string> subSubFields = {"Access-Latency", "Size"};
+  nodeChecker<uint16_t>(configFile_[root][subFields[0]], subFields[0],
+                        std::make_pair(1, UINT16_MAX), ExpectedValue::UInteger,
+                        64);
+  nodeChecker<uint16_t>(configFile_[root][subFields[1]][subSubFields[0]],
+                        subSubFields[0], std::make_pair(1, UINT16_MAX),
+                        ExpectedValue::UInteger, 1);
+  // Default L1-Data size is 1024 * 32 = 32 KiB
+  uint16_t cacheLineLen = configFile_[root][subFields[0]].as<uint16_t>();
+  nodeChecker<uint64_t>(
```

Comment:
Should we allow such large L1 cacheSizes? surely uint16_t will suffice?

Reply:
Yes, something like the M1 firestorm core has 128KiB of L1 Data cache per core